carbon/feed.py

```diff
@@ -0,0 +1,341 @@
+from abc import ABC, abstractmethod
+from collections.abc import Generator
+from contextlib import closing
+from datetime import datetime
+from typing import IO, Any, ClassVar
+
+from lxml import etree as ET
+from sqlalchemy import func, select
+from sqlalchemy.sql.selectable import Select
+
+from carbon.database import DatabaseEngine, aa_articles, dlcs, orcids, persons
+from carbon.helpers import (
+    get_group_name,
+    get_hire_date_string,
+    get_initials,
+)
+
+
+class BaseXmlFeed(ABC):
+    """Base XML feed class.
+
+    This is the abstract class for creating XML feeds. The following class attributes
+    are unique per subclass of carbon.feed.BaseXmlFeed:
+
+        root_element_name: The 'tag' assigned to the root Element.
+        query: The select statmenet submitted to the Data Warehouse to retrieve records.
+
+    Attributes:
+        engine: A configured carbon.database.DatabaseEngine that can connect to the
+            Data Warehouse.
+        output_file: A file-like object (stream) into which normalized XML strings
+            strings are written.
+
+    """
+
+    root_element_name: str | ET.QName = ""
```

Comment:
Would it easier to just convert `ET.QName` to `str` before passing it as a parameter?

Reply:
Good idea! Applied the change.